scripts/zkai_paper_claim_pack_gate.py

```
@@ -0,0 +1,354 @@
+    cases.append(("unknown_field_injection", mutated))
+
+    return cases
+
+
+def write_json(path: pathlib.Path, payload: dict[str, Any]) -> None:
+    validate_payload(payload)
+    if path.exists() and path.is_symlink():
+        raise ClaimPackGateError("output path must not be a symlink")
+    path.parent.mkdir(parents=True, exist_ok=True)
+    if path.parent.is_symlink():
+        raise ClaimPackGateError("output parent must not be a symlink")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)